### PR TITLE
[LIQ-626] Refactor sdk to support v2 CandyShop program

### PR DESCRIPTION
**Pull request by nibbus0x**

### CODE REVIEW

File: core/cli/src/cli.ts [hunk 1]
```diff
@@ -1,13 +1,20 @@
...
+  loadTokenAccountMints,
+  findAssociatedTokenAddress
+} from './helper/account';
 
 const CMD = new Command();
 
 CMD.description('CLI for interacting with Candy Shop');
 
 function programCommand(name: string) {
+  // Should we add required option here to specify program version?
```
Comment (RustySol) [inline core/cli/src/cli.ts:17]: can do in upcoming PR

File: core/sdk/src/CandyShop.ts [hunk 2]
```diff
@@ -61,11 +73,13 @@ import {
...
   CandyShopBuyNowParams,
-  CandyShopUpdateParams
+  CandyShopUpdateParams,
+  CandyShopConstructorParams,
+  CandyShopVersion
 } from './CandyShopModel';
 import { configBaseUrl } from './config';
 import { CandyShopError, CandyShopErrorType } from './utils';
-import { updateCandyShop } from './api/program/marketplace/updateCandyShop';
+import { updateCandyShop } from './api/program/v1/marketplace/updateCandyShop';
```
Comment (RustySol) [inline core/sdk/src/CandyShop.ts:82]: do we need v2 for this also?

Comment (nibbus0x) [reply core/sdk/src/CandyShop.ts]: Good catch, updated

**Review by RustySol — APPROVED** [review]
LGTM

**Pull request merged**